test/prism_regression/namer_arguments.rb

```diff
@@ -0,0 +1,10 @@
+# typed: true
+class A
+  def take_arguments(a, b=1, *c, d:, e:2, **f, &g)
+    [a,b,c,d,e,f,g]
+    h = 1
+    proc do |a, b=1, *c, d:, e:2, **f, &g; h|
+      [a,b,c,d,e,f,g,h]
+    end
+  end
+end
```

Comment:
Does this test capture the issue? I added it to `prism` branch and it passed without the patch 🤔